actionview/test/template/test_case_test.rb

```diff
@@ -382,6 +382,62 @@ def page
     end
   end
 
+  class HTMLDecoderTest < ActionView::TestCase
+    test "rendered_html is a Nokogiri::XML::Element" do
+      developer = DeveloperStruct.new("Eloy")
+
+      render "developers/developer", developer: developer
+
+      assert_kind_of Nokogiri::XML::Element, rendered_html
+      assert_equal developer.name, document_root_element.text
+    end
+
+    test "do not memoize the rendered_html in view tests" do
+      concat form_tag("/foo")
+
+      assert_equal "/foo", document_root_element.at("form")["action"]
+
+      concat content_tag(:b, "Strong", class: "foo")
+
+      assert_equal "/foo", document_root_element.at("form")["action"]
+      assert_equal "foo", document_root_element.at("b")["class"]
+    end
+  end
+
+  class JSONDecoderTest < ActionView::TestCase
+    test "rendered_json is an ActiveSupport::HashWithIndifferentAccess" do
+      developer = DeveloperStruct.new("Eloy")
+
+      render formats: :json, partial: "developers/developer", locals: { developer: developer }
+
+      assert_kind_of ActiveSupport::HashWithIndifferentAccess, rendered_json
+      assert_equal developer.name, rendered_json[:name]
+    end
+  end
+
+  class MethodNameCollisionTest < ActionView::TestCase
+    def rendered_html
+      :already_defined
+    end
+
+    test "rendered_html does not override an existing helper" do
+      assert_equal :already_defined, rendered_html
+    end
```

Comment:
@zzak I've added test coverage to make sure that the generated methods (like `rendered_html`) won't re-define user-land defined helper methods.

Does that cover what you had in mind in https://github.com/rails/rails/pull/49194#discussion_r1328157681?